pkg/daemon/daemon.go

```diff
@@ -1629,3 +1634,71 @@ func forceFileExists() bool {
 
 	return false
 }
+
+// patchLabelsOnNode takes a map of key/value string pairs and patches them as labels on the node
+func (dn *Daemon) patchLabelsOnNode(labels map[string]interface{}) error {
+	var err error
+	var patchData []byte
+	patch := struct {
+		Metadata map[string]interface{} `json:"metadata"`
+	}{
+		Metadata: map[string]interface{}{
+			"labels": labels,
+		},
+	}
+	dn.logSystem("Patching labels %v", labels)
+	patchData, err = json.Marshal(&patch)
+	if err != nil {
+		dn.logSystem("Error in patching labels: %v", err)
+		return err
+	}
+	_, err = dn.kubeClient.CoreV1().Nodes().Patch(
+		context.TODO(), dn.name, types.MergePatchType, patchData, metav1.PatchOptions{})
+	if err != nil {
+		dn.logSystem("Error patching labels: %v", err)
+		return err
+	}
+	return nil
+}
+
+// Add or remove excludeFromLoadBalancerLabel to the node. If addLabel=true, it
+// adds the label if the label doesn't already exist and sets it to a custom value
+// so as to know that it was added by MCO. If addLabel=false, it removes the label,
+// only if its value is the known custom value, otherwise it leaves it as is.
+func (dn *Daemon) addOrRemoveExcludeFromLoadBalancerLabel(addLabel bool) error {
+	var err error
+	var labelValue interface{}
+	// Add the label only if not present; remove the label only if it has our custom value
+	labelValue, labelExists := dn.node.Labels[excludeFromLoadBalancerLabel]
+	if addLabel {
+		if labelExists {
+			// TODO remove log line below and make the if condition more compact
+			dn.logSystem("node %s already has label %s=%s (addLabel=%t), nothing to add",
+				dn.name, excludeFromLoadBalancerLabel, labelValue, addLabel)
+			return nil
+		}
+		labelValue = excludeFromLoadBalancerLabelValue
+	} else {
+		if labelExists {
+			if labelValue != excludeFromLoadBalancerLabelValue {
+				dn.logSystem("label %s=%s was not added by MCO, not removing it",
+					excludeFromLoadBalancerLabel, labelValue)
+				return nil
+			}
+		} else {
+			dn.logSystem("label %s not present on node %s, nothing to remove",
+				excludeFromLoadBalancerLabel, dn.name)
+			return nil
+		}
+
+		// removing the label by setting its value to nil
+		labelValue = nil
+	}
+	err = dn.patchLabelsOnNode(map[string]interface{}{excludeFromLoadBalancerLabel: labelValue})
```

Comment:
Not sure @deads2k older comment was addressed here yet:
`for any other error on the stat, I would expect no action from this function.`